### PR TITLE
EUI-5094 Hearing stage and result page

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-5094

### Change description ###
Hearings actuals-Display 'Hearing stage and result' page 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: api/hearings/models/hearings.enum.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export enum EXUIDisplayStatusEnum {
   FAILURE = 'FAILURE',
   CANCELLATION_REQUESTED = 'CANCELLATION REQUESTED',
   VACATED = 'VACATED',
-  AWAITING_ACTUALS = 'AWAITING ACTUALS',
+  AWAITING_ACTUALS = 'AWAITING HEARING ACTUALS',
```
Comment (paddybasi) [inline api/hearings/models/hearings.enum.ts:51]: already done in https://github.com/hmcts/rpx-xui-webapp/pull/1773

File: src/cases/components/case-hearings-list/case-hearings-list.component.html [hunk 2]
```diff
@@ -25,11 +25,12 @@
     </td>
     <td class="govuk-table__cell govuk-table-width-30-percent">
       <div *ngIf="hasUpdateAction" class="div-action">
-        <a href="javascript:void(0)" (click)="viewAndEdit()">View or edit hearing</a>&nbsp;&nbsp;&nbsp;&nbsp;
+        <ng-container *ngIf="!hasAddEdit(hearing)"><a href="javascript:void(0)" (click)="viewAndEdit()">View or edit</a>&nbsp;&nbsp;&nbsp;&nbsp;</ng-container>
```
Comment (paddybasi) [inline src/cases/components/case-hearings-list/case-hearings-list.component.html:28]: already done in https://github.com/hmcts/rpx-xui-webapp/pull/1773

File: api/hearings/models/hearings.enum.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply api/hearings/models/hearings.enum.ts]: @paddybasi - this is now outdated as I merged the changes from base branch after you merged your changes (EUI-5092).

File: src/cases/components/case-hearings-list/case-hearings-list.component.html (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply src/cases/components/case-hearings-list/case-hearings-list.component.html]: @paddybasi - this is now outdated as I merged the changes from base branch after you merged your changes (EUI-5092).

File: api/hearings/models/hearingStatusMappings.ts [hunk 3]
```diff
@@ -45,7 +45,7 @@ export const hearingStatusMappings: HearingStatusMapping[] = [
   },
   {
     hmcStatus: HMCStatus.AWAITING_ACTUALS,
-    exuiSectionStatus: EXUISectionStatusEnum.PAST_AND_CANCELLED,
+    exuiSectionStatus: EXUISectionStatusEnum.UPCOMING,
```
[the remark this answers is not in the document]
Comment (johnbenjamin-hmcts) [reply api/hearings/models/hearingStatusMappings.ts]: @paddybasi - this is now outdated as I merged the changes from base branch after you merged your changes (EUI-5092).

File: api/prd/lov/data/lov.mock.data.ts [hunk 4]
```diff
@@ -160,6 +160,114 @@ export const DEFAULT_HEARING_CANCEL_REF: LovRefDataModel[] = [
...
+  {
+    key: 'postponedOtherPartyUnableToAttend',
+    value_en: 'Postponed, Other Party unable to attend',
+    value_cy: '',
+    hintText_EN: 'Postponed, Other Party unable to attend',
+    hintTextCY: '',
+    order: 4,
+    parentKey: null,
+  },
+];
```
Comment (rayliangatos) [inline api/prd/lov/data/lov.mock.data.ts:239]: Could you confirm with BA and see if this is still matched with requirement?
Are you refering to this prototype https://ucd-xui2-prototype.herokuapp.com/hearing-actuals/version3/actuals-details or others?

File: src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                  </div>
+                </div>
+                <div class="govuk-radios__item">
+                  <input class="govuk-radios__input" id="cancelled" formName="hearingResult" name="hearingResult"
+                    type="radio" value="cancelled" data-aria-controls="conditional-cancelled" (change)="onHearingResult(hearingResultEnum.CANCELLED)">
+                  <label class="govuk-label govuk-radios__label" for="cancelled">
+                    Withdrawn, cancelled or vacated
+                  </label>
+                </div>
+                <div class="govuk-radios__conditional" *ngIf="hearingResultType === hearingResultEnum.CANCELLED">
```
Comment (rayliangatos) [inline src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.html:69]: could this change to NgSwitchCase?

File: src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { Observable, Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { HearingActualsStateData } from 'src/hearings/models/hearingActualsStateData.model';
```
Comment (rayliangatos) [inline src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts:6]: src

File: api/prd/lov/data/lov.mock.data.ts (hunk 4 above)
Comment (johnbenjamin-hmcts) [reply api/prd/lov/data/lov.mock.data.ts]: @rayliangatos thanks for bringing this up in today's stand-up. I have now removed this code and made necessary changes as dropdown is not required for the first radio button option.

File: src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts (hunk 6 above)
Comment (johnbenjamin-hmcts) [reply src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts:6]: I have now updated the path.

File: src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.html (hunk 5 above)
Comment (johnbenjamin-hmcts) [reply src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.html]: @rayliangatos - I think NgSwitchCase is not possible in this scenario as each dropdown is a child of radio button and we have to show all the radio buttons by default and show/hide dropdowns based on radio button selection.

File: api/prd/lov/data/lov.mock.data.ts [hunk 7]
```diff
@@ -160,6 +160,84 @@ export const DEFAULT_HEARING_CANCEL_REF: LovRefDataModel[] = [
...
```
Comment (rayliangatos) [inline api/prd/lov/data/lov.mock.data.ts:209]: Coud we add more options as per the prototype did? And the order is incremental please

**Review by rayliangatos — APPROVED** [review]
thanks for addressing the comments.

File: src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  public ngOnInit(): void {
+    this.hearingStageResultForm = this.formBuilder.group({
+      hearingStage: [''],
+      hearingResult: ['', Validators.required],
+      adjournedReason: [''],
+      cancelledReason: ['']
+    });
+    this.hearingTypes$ = this.lovRefDataService.getListOfValues('HearingType', 'SSCS');
```
Comment (paddybasi) [inline src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts:46]: could you remove these please in favour of using `RefDataResolver` as the serviceId is hardcoded

**Review by paddybasi — CHANGES_REQUESTED** [review]

File: src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts (hunk 8 above)
Comment (rayliangatos) [reply src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts:46]: that is good spot, thanks @paddybasi

Comment (johnbenjamin-hmcts) [reply src/hearings/containers/hearing-actuals/hearing-stage-result/hearing-stage-result.component.ts]: @paddybasi I have now updated the code as we discussed.

**Review by paddybasi — APPROVED** [review]

**Pull request merged**